#163519099 login page

Nziranziza opened the pull request:
#### What does this PR do?
add login functionality
#### Description of Task to be completed?
- Write Unit tests
- Add login reducers
- Add login Actions

#### How should this be manually tested?
- Git clone
- npm install
- npm start
- Navigate to /login

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#163519099

#### Screenshots (if appropriate)

#### Questions:

filschristian committed:
feat(login-page): first commit

filschristian committed:
feat(login-page): first commit

Nziranziza committed:
feat(login) front-end login
- add login reducers
- add login actions
- add login UI component
[starts #163519099]

Nziranziza committed:
feat(login) front-end login
- add login reducers
- add login actions
- add login UI component
[starts #163519099]

Nziranziza committed:
feat(login) front-end login
- git rebase
[ continues #163519099]

Nziranziza committed:
feat(login) front-end login
- fixed rebase errors
- [continues #163519099]

Nziranziza committed:
feat(login) front-end login
- fixed form-errors
- [continues #163519099]

Nziranziza committed:
feat(login) front-end login
- change method to arrow finction

Nziranziza committed:
feat(login) front-end login
- add submit action
- add submit login form reducer
- modify login button to show loading spin
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- remove routes
- [continues #163519099]

Nziranziza committed:
feat(login)
removed unnecessary console
- change function name
[continues #163519099]

filschristian committed:
feat(login): add component props validation

Nziranziza committed:
feat(login) front-end login
- add test for login reducers
- add test for currentUser reducers
- add test for currentUser actions
- add test for user reducer
- add test for user action
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- modify folder structure
[continues #163519099]

filschristian committed:
feat(login): restructure the project

filschristian committed:
feat(login): fix conflicts and add error handling

Nziranziza committed:
feat(login) front-end login
- modify reducer

Nziranziza committed:
feat(login) front-end login
- add test for login action
[continues #163519099]

kimotho-njoki commented on src/components/Auth/Login.jsx:
```diff
@@ -1,7 +1,141 @@
+
+  render() {
+    const {
+      credentials: { username, password },
```
why is there a trailing comma at the end?

kimotho-njoki commented on src/components/Auth/Login.jsx:
```diff
@@ -1,7 +1,141 @@
+    const {
+      credentials: { username, password },
+    } = this.props;
+    const { submitting } = this.props;
```
If `credentials: { username, password }` and `submitting` are both `this.props` why are they on different lines?

kimotho-njoki commented on src/components/Auth/index.js:
```diff
@@ -0,0 +1,21 @@
+import Login from './Login';
+
+class AuthComponent extends Component {
+  constructor(props) {
```
why do we need this constructor or even an empty state in this case?

kimotho-njoki commented on src/redux/actions/loginActions.js:
```diff
@@ -0,0 +1,94 @@
+        return;
+      }
+      dispatch({
+        type: LOGIN_FAILURE,
```
why are we handling catch twice? whats the need of the `if` statement

kimotho-njoki commented on src/redux/reducers/currentUserReducer.js:
```diff
@@ -0,0 +1,19 @@
+const currentUserReducer = (state = initialState, { type, payload }) => {
+  switch (type) {
+    case SET_CURRENT_USER:
+      return { ...state, profile: payload, isLoggedIn: true };
```
why is the return in a single line?

kimotho-njoki commented on src/redux/reducers/currentUserReducer.js:
```diff
@@ -0,0 +1,19 @@
+    case SET_CURRENT_USER:
+      return { ...state, profile: payload, isLoggedIn: true };
+    default:
+      return { ...state };
```
there's no need for braces, you can just write `return ...state` since its a single return

kimotho-njoki commented on src/redux/reducers/loginReducer.js:
```diff
@@ -0,0 +1,37 @@
+    case INPUT_VALIDATION_SUCCESS:
+      return { ...state, error: undefined };
+    default:
+      return { ...state };
```
If it's a single return statement, get rid of the brackets. If it's multiple return values, separate them on different lines.

kimotho-njoki commented on src/redux/reducers/userReducer.js:
```diff
@@ -1,17 +1,20 @@
+      };
     default:
-      return state;
+      return { ...state };
```
remove!

kimotho-njoki commented on src/redux/reducers/userReducer.js:
```diff
@@ -1,17 +1,20 @@
-import { LOGIN } from '../actions-types/userTypes';
+import { SET_PROFILE } from '../actions-types';
 
 const initialState = {
```
Do we have a separate file for `initial state`?

Nziranziza committed:
feat(login) front-end login
- add redirect on successfuly login
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- switch from axios to fetchAPI
[continues #163519099]

filschristian committed:
feat(login): add tests

Nziranziza committed:
feat(login) front-end login
- add login action test
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- add more login action test

[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- add more login action test

[continues #163519099]

filschristian commented on src/__tests__/actions/loginActions.test.js:
```diff
@@ -0,0 +1,203 @@
+  beforeEach(() => {
+    store = mockStore({});
+  });
+  test('should return filed and its value', () => {
```
change filed to field

Nziranziza commented on src/__tests__/reducers/loginReducer.test.js:
```diff
@@ -0,0 +1,150 @@
+    expect(res.error).toBeUndefined();
+  });
+
+  test('should previous state when no action type provided', () => {
```
add return between should and previous

Nziranziza commented on src/components/Auth/Login.jsx:
```diff
@@ -1,7 +1,150 @@
+            />
+          </div>
+          <div className="align-right" id="forget-psswd">
+            <a href="forgot-password-reset.html">Forget Password?</a>
```
use link instead of a

Nziranziza commented on src/components/Auth/Login.jsx:
```diff
@@ -1,7 +1,150 @@
+          <Button
+            type="button"
+            classes={`primary color-white ${submitting ? 'loading' : ''}`}
+            color-white
```
remove color-white

filschristian committed:
feat(login ui): implement feedback from PR review

- restructure several default returns in reducers
- add a single file for the initial state
- refactor reducers to adapt to new changes
- refactor tests to adapt to new changes

[Starts #163519099]

filschristian committed:
feat(login): change the initial state from js to json

filschristian committed:
feat(login): change import of initial state object

filschristian committed:
feat(login): add redirection to home page after successfull login

Nziranziza committed:
feat(login) front-end login
- add social login icon component
- add unit test for social login icon
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- git pull

filschristian committed:
feat(login): implement feedback

filschristian committed:
feat(login): add unit tests

filschristian committed:
feat(login): add more unit tests

Nziranziza committed:
feat(login) front-end login
- add more test
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- add more test
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- add more test
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- add more test
[continues #163519099]

Nziranziza committed:
feat(login) front-end login
- add functionality to save token in local storage
[continues #163519099]

Nziranziza committed:
feat(login) PO's feedback
- change error message
[finishes #163519099]
